### PR TITLE
[Card] Adjust padding of Card.Section, footer and header

### LINKED ISSUE #954: [Card] Padding issue between footer actions and subdued section card
# Issue summary

When using a subdued card and a card with footer actions the footer actions area butts 🍑 up uncomfortably against the subdued card section.

## Expected behavior

There should be some padding added to the top of the footer section to ease the eyes.

## Actual behavior

much gross

## Steps to reproduce the problem

1. Paste this code in le playground for a visual
```
import * as React from 'react';
import {Page, Card, List} from '../src';

interface State {}

export default class Playground extends React.Component<{}, State> {
  render() {
    return (
      <Page title="Playground">
        <Card
          secondaryFooterAction={{content: 'Dismiss'}}
          primaryFooterAction={{content: 'Export Report'}}
        >
          <Card.Section title="Deactivated reports" subdued>
            <List>
              <List.Item>Payouts</List.Item>
              <List.Item>Total Sales By Channel</List.Item>
            </List>
          </Card.Section>
        </Card>
      </Page>
    );
  }
}
```

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 3.6.0
- Browser: chrome
- Device: mac
- Operating System: macOS 10.14

### PR DESCRIPTION
**Pull request by elizabethletourneau**
### WHY are these changes introduced?

Resolves #954  

When a subdued section `Card` and a footer prop were combined there was no padding/margin between the `Button Group` and the edge of the subdued card. 
It was also discovered in the interim there was the same issue with the Header  button Group and a subdued section card as well so this will be resolved.

### WHAT is this pull request doing?

This request is adding a margin to the footer section.
Before: 
image

After: 
image

Small screen:
image

With no header and regular section cards:
image

Small screen:
image
<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page, Card, List, Popover, Button, ActionList} from '../src';

interface State {}

export default class Playground extends React.Component<{}, State> {
  render() {
    return (
      <Page title="Playground">
        <Card
          secondaryFooterAction={{content: 'Dismiss'}}
          primaryFooterAction={{content: 'Export Report'}}
        >
          <Card.Header
            actions={[
              {
                content: 'Preview',
              },
            ]}
            title="Staff accounts"
          >
            <Popover
              active
              activator={
                <Button disclosure plain>
                  Add account
                </Button>
              }
              onClose={() => {}}
            >
              <ActionList items={[{content: 'Member'}, {content: 'Admin'}]} />
            </Popover>
          </Card.Header>
          <Card.Section title="Deactivated reports" subdued>
            <List>
              <List.Item>Payouts</List.Item>
              <List.Item>Total Sales By Channel</List.Item>
            </List>
          </Card.Section>
        </Card>
      </Page>
    );
  }
}
```

</details>

<details>
<summary>without subdued section:</summary>

```jsx
import * as React from 'react';
import {Page, Card, List, Popover, Button, ActionList} from '../src';

interface State {}

export default class Playground extends React.Component<{}, State> {
  render() {
    return (
      <Page title="Playground">
        <Card
          secondaryFooterAction={{content: 'Dismiss'}}
          primaryFooterAction={{content: 'Export Report'}}
        >
          <Card.Header
            actions={[
              {
                content: 'Preview',
              },
            ]}
            title="Staff accounts"
          >
            <Popover
              active
              activator={
                <Button disclosure plain>
                  Add account
                </Button>
              }
              onClose={() => {}}
            >
              <ActionList items={[{content: 'Member'}, {content: 'Admin'}]} />
            </Popover>
          </Card.Header>
          <Card.Section title="Deactivated reports">
            <List>
              <List.Item>Payouts</List.Item>
              <List.Item>Total Sales By Channel</List.Item>
            </List>
          </Card.Section>
        </Card>
      </Page>
    );
  }
}
```

</details>
### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Place your settings in this file to overwrite default and user settings.
 {
   "css.validate": false,
-  "editor.formatOnSave": true,
+  "editor.formatOnSave": false,
```
Comment (ry5n) [inline .vscode/settings.json]: This change seems unrelated to the PR?

File: UNRELEASED.md [hunk 2]
```diff
@@ -14,6 +14,8 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Bug fixes
 
+- Add margin top to `Card`'s footer to separate `Button Group` ([#962](https://github.com/Shopify/polaris-react/pull/962))
```
Comment (solonaarmstrong) [inline UNRELEASED.md]:
```suggestion
- Added top margin to the footer of `Card` to separate `Button Group` ([#962](https://github.com/Shopify/polaris-react/pull/962))
```

'- Past tense verb '

https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md#component-name-formats

File: .vscode/settings.json (hunk 1 above)
Comment (solonaarmstrong) [inline .vscode/settings.json]: 🔥

**Comment by solonaarmstrong** [comment]
Looks 👌 

This adds a margin between the footer and a regular section, as well, so I would just make sure that a designer (@ry5n ?) agrees with that decision. It looks better to me with the extra space for both the subdued and regular sections, but I don't know if there was a reason behind the lack of top margin for regular sections.

**Comment by solonaarmstrong** [comment]
Given there doesn't seem to be an elegant solution to adding a top margin only on the footer when the final section is subdued, what was the decision? Did we agree to add the top margin across the board?

File: UNRELEASED.md [hunk 3]
```diff
@@ -14,6 +14,8 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Bug fixes
 
+- Adjusted padding to `Card`'s sections, header, and footer to separate actions in header and footer actions in footer ([#962](https://github.com/Shopify/polaris-react/pull/962))
```
Comment (ry5n) [inline UNRELEASED.md]: This could be clearer. Right now I understand that padding changed in several places in Card, but not getting what really changed and why.

File: src/components/Card/Card.scss [hunk 4]
```diff
@@ -20,7 +20,7 @@
   padding: spacing() spacing() 0;
 
   @include page-content-when-not-fully-condensed {
-    padding: spacing(loose) spacing(loose) 0;
+    padding: spacing(loose) spacing(loose) spacing();
```
Comment (ry5n) [inline src/components/Card/Card.scss]: Don’t forget that this change to bottom padding needs to be applied on line 20, for small screens, as well.

In fact, it might be good to write it like this:

```scss
.Header {
  // Shorthand is good for the default breakpoint
  padding: spacing();

  // At other breakpoints, overriding only the specific rules that change makes it easier 
  // to understand intent and can avoid things becoming out of sync across breakpoints
  @include page-content-when-not-fully-condensed {
    padding-top: spacing(loose);
    padding-right: spacing(loose);
    padding-left: spacing(loose);
```

File: src/components/Card/Card.scss [hunk 5]
```diff
@@ -42,6 +42,11 @@
 
 .Section-subdued {
   background-color: color('sky', 'lighter');
+  border-top: border-width() solid color('sky');
+
+  + .Footer {
```
Comment (solonaarmstrong) [inline src/components/Card/Card.scss]: Oh, of course! This is 👌

File: src/components/Card/Card.scss [hunk 6]
```diff
@@ -17,18 +20,24 @@
 }
 
 .Header {
-  padding: spacing() spacing() 0;
+  padding-top: spacing();
+  padding-right: spacing();
+  padding-left: spacing();
+  padding-bottom: $header-spacing;
```
Comment (solonaarmstrong) [inline src/components/Card/Card.scss]: Is there a reason this isn't using `padding`?

```suggestion
padding: spacing() spacing() $header-spacing spacing();
```

File: src/components/Card/Card.scss [hunk 7]
```diff
@@ -17,18 +20,24 @@
...
 
   @include page-content-when-not-fully-condensed {
-    padding: spacing(loose) spacing(loose) 0;
+    padding-top: spacing();
+    padding-right: spacing(loose);
+    padding-left: spacing(loose);
+    padding-bottom: $header-spacing;
   }
```
Comment (solonaarmstrong) [inline src/components/Card/Card.scss]: Same question here.

File: src/components/Card/Card.scss [hunk 8]
```diff
@@ -61,9 +71,19 @@
 .Footer {
   display: flex;
   justify-content: flex-end;
-  padding: 0 spacing() spacing();
+  padding-top: $footer-spacing;
+  padding-right: spacing();
+  padding-left: spacing();
+  padding-bottom: spacing();
```
Comment (solonaarmstrong) [inline src/components/Card/Card.scss]: Same question here.

File: src/components/Card/Card.scss [hunk 9]
```diff
@@ -61,9 +71,19 @@
...
 
   @include page-content-when-not-fully-condensed {
-    padding: 0 spacing(loose) spacing(loose);
+    padding-top: $footer-spacing;
+    padding-right: spacing(loose);
+    padding-left: spacing(loose);
+    padding-bottom: spacing();
```
Comment (solonaarmstrong) [inline src/components/Card/Card.scss]: Same question here.

File: src/components/Card/Card.scss (hunk 6 above)
Comment (elizabethletourneau) [reply src/components/Card/Card.scss]: Ryan commented to split out the other larger screen view to make it easier to understand which paddings had changed between the two views. I shouldn't have changed the mobile view one though so I have changed those back.

File: src/components/Card/Card.scss [hunk 10]
```diff
@@ -17,18 +20,21 @@
...
 .Header {
-  padding: spacing() spacing() 0;
+  padding-top: spacing() spacing() $header-spacing spacing();
 
   @include page-content-when-not-fully-condensed {
-    padding: spacing(loose) spacing(loose) 0;
+    padding-top: spacing();
+    padding-right: spacing(loose);
+    padding-left: spacing(loose);
+    padding-bottom: $header-spacing;
```
[the remark this answers is not in the document]
Comment (ry5n) [reply src/components/Card/Card.scss]: If you do it that way, this value will only appear once, and there’s no longer a good reason to use a variable for it.

File: .vscode/settings.json (hunk 1 above)
Comment (ry5n) [inline .vscode/settings.json]: ^^

File: UNRELEASED.md [hunk 11]
```diff
@@ -17,6 +17,7 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 ### Bug fixes
 
 - Fixed the `DataTable` sort direction not reversing on second sort of the initially sorted column ([#918](https://github.com/Shopify/polaris-react/pull/918)) (thanks [@tabrez96](https://github.com/tabrez96) for the [issue report](https://github.com/Shopify/polaris-react/issues/873))
+- Adjusted top and bottom padding to the header, footer and sections in `Card` to add space between action buttons in the header and footer and the card sections. ([#962](https://github.com/Shopify/polaris-react/pull/962))
```
Comment (ry5n) [inline UNRELEASED.md]: 👍

File: src/components/Card/Card.scss [hunk 12]
```diff
@@ -17,18 +17,19 @@
 }
 
 .Header {
-  padding: spacing() spacing() 0;
+  padding: spacing() spacing() rem(12px) spacing();
```
Comment (solonaarmstrong) [inline src/components/Card/Card.scss]: This is not the first time I've seen the need for `12px` and I wonder if we need to add base * 3 in our spacing Sass map?

@kaelig Is this something you'd have an opinion about?

Comment (ry5n) [reply src/components/Card/Card.scss]: I dislike our spacing API primarily because it actively works against a shared vocabulary between designers and devs. Design tools don’t allow named spacing values, but our spacing API enforces it.

(I should also point out that other values fit the system—24px is a valid spacing value, same with 28, and 36, and 40…)

**Comment by solonaarmstrong** [comment]
👌

File: src/components/Card/Card.scss (hunk 12 above)
Comment (elizabethletourneau) [reply src/components/Card/Card.scss]: Resolving this. Opened an issue for discussion [here](https://github.com/Shopify/polaris-react/issues/1007).
